syntax = "proto3";

package influence_service;

service InfluenceService {
    rpc ComputeInfluences (stream InfluenceRequest) returns (InfluenceResponse);
}

message InfluenceRequest {
    bytes train_data = 1;
    bytes model = 2;
    bytes test_data = 3;
    bytes train_labels = 4;
    bytes test_labels = 5;
    int32 num_influential = 6;
    bytes untransformed_train = 7;
    bytes untransformed_train_labels = 8;
}

message InfluenceResponse {
    repeated float influences = 1;
    bytes positive = 2;
    bytes negative = 3;
}